deploy/docker/notebook/base/base.yaml

```diff
@@ -421,26 +425,92 @@ addfiles:
+              """Local spawner that runs single-user servers as the same user as the Hub itself.
+              Overrides user-specific env setup with no-ops.
+              """
 
-              logging.getLogger().info(f"Saved notebook at {os_path}")
-              try:
-                    r = requests.post("https://polus-notebooks-hub-backend.ci.aws.labshare.org/file-data/new-file", json={"user": os.getenv("JUPYTERHUB_USER"), "filePath": os_path}, timeout=1)
-                    if r.status_code == 200:
-                        logging.getLogger().info(f"Registered {os_path} with Notebooks Hub")
-                    else:
-                        logging.getLogger().error(f"Could not register {os_path} with Notebooks Hub. Return code: {r.status_code}")
-              except Exception as e:
-                    logging.getLogger().error(f"Could not register {os_path} with Notebooks Hub", exc_info=e)
+              def is_inside_path(self, path_to_test, path):
+                  return os.path.samefile(os.path.commonpath([path, path_to_test]), path)
               
-          
-          c.FileContentsManager.post_save_hook = script_post_save
\ No newline at end of file
+              def script_post_delete(self, os_path):
+
+                  hookVar = hookVariables()
+
+                  os_path = str(Path().resolve()) + os_path
+                  
+                  if not (self.is_inside_path(os_path, hookVar.work_path) or self.is_inside_path(os_path, hookVar.shared_path)):
+                      return
+                  
+                  try:
+                      r = requests.post(hookVar.POLUS_NOTEBOOKS_HUB_API+"/file-data/delete-file", json={"user": os.getenv("JUPYTERHUB_USER"), "filePath": os_path}, timeout=1)
+                  except Exception as e:
+                      if os.environ["POLUS_NOTEBOOKS_HUB_FILE_LOGGING_ENABLED"] == "True":
+                          logging.getLogger().error(f"Failed to delete {os_path} from Notebooks Hub", exc_info=e)
+
+              def script_post_save(self, os_path):
+                  
+                  hookVar = hookVariables()
+
+                  os_path = str(Path().resolve()) + os_path
+
+                  if "jovyan" not in os_path:
+                      return
+
+                  if not (self.is_inside_path(os_path, hookVar.work_path) or self.is_inside_path(os_path, hookVar.shared_path)):
+                      return
+
+                  try:
+                      r = requests.post(hookVar.POLUS_NOTEBOOKS_HUB_API+"/file-data/new-file", json={"user": os.getenv("JUPYTERHUB_USER"), "filePath": os_path}, timeout=1)
+                  except Exception as e:
+                      if os.environ["POLUS_NOTEBOOKS_HUB_FILE_LOGGING_ENABLED"] == "True":
+                          logging.getLogger().error(f"Failed to register {os_path} with Notebooks Hub", exc_info=e)
+
+              def script_post_rename(self, os_old_path, os_new_path):
```

Comment:
Does rename hook get called when the file is moved in JupyterLab? If so, we need to consider the logic of what happens when it's moved in or out of our tracked folders `work_path` and `shared_path`. 

For example, user created a notebook inside a folder in their home folder (which acts like scratch space), then decided to keep it, so moved it to `work_path. In this scenario, the file did not exist in your database before, so it needs to be created. Therefore, `/file-data/new-file` needs to be called instead of `/file-data/rename-file`. There are more scenarios like that that need to be accounted for here.

Reply:
In my mongodb API, all the rename file API does is call file-delete and file-new essentially and checks to see if they exists first before running the create/update or delete command. So this scenario and most other edge cases should be handled.